provider/parameter.go

```diff
@@ -134,6 +134,10 @@ func parameterDataSource() *schema.Resource {
 				return diag.Errorf("parameter can't be immutable and ephemeral")
 			}
 
+			if !parameter.Optional && parameter.Ephemeral {
+				return diag.Errorf("ephemeral parameter requires the default property")
```

Comment:
Sanity check: do parameters have an optional field or is it inferred by setting default, and is that the only way to do it? (I always stop to ask when I see the check and message say something different.)

Reply:
> do parameters have an optional field

No.

> is it inferred by setting default

Yes 👍 

> is that the only way to do it

Yes 👍